test/e2e/operator_test.go

```diff
@@ -758,6 +759,118 @@ func TestScopeChange(t *testing.T) {
 	}
 }
 
+// TestAllowedSourceRanges creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type and verifies that the
+// operator correctly configures the associated load balancer and updates its
+// configuration when the ingresscontroller's allowed source ranges are changed.
+func TestAllowedSourceRanges(t *testing.T) {
+	platform := infraConfig.Status.Platform
+
+	supportedPlatforms := map[configv1.PlatformType]struct{}{
+		configv1.AWSPlatformType:   {},
+		configv1.AzurePlatformType: {},
+		configv1.GCPPlatformType:   {},
+	}
+	if _, supported := supportedPlatforms[platform]; !supported {
+		t.Skipf("test skipped on platform %q", platform)
+	}
+
+	t.Log("Creating ingresscontroller with spec.endpointPublishingStrategy.loadBalancer.allowedSourceRanges set to 127.0.0.0/8")
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "sourcerange"}
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newLoadBalancerController(name, domain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		AllowedSourceRanges: []operatorv1.CIDR{"127.0.0.0/8"},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, nondefaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+
+	if isServiceInternal(lbService) {
+		t.Fatalf("load balancer is internal but should be external")
+	}
+
+	n := 1
+	checkRoute := func() (string, error) {
+		// Use a unique host name for each request to work around
+		// caching of negative responses.
+		url := fmt.Sprintf("http://x%d.%s./", n, domain)
+		n++
+
+		request, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return "", err
+		}
+
+		client := &http.Client{Timeout: 5 * time.Second}
+		response, err := client.Do(request)
+		if err != nil {
+			return "", err
+		}
+
+		_, err = io.Copy(ioutil.Discard, response.Body)
+		response.Body.Close()
+		if err != nil {
+			return "", err
+		}
+
+		return response.Status, nil
+	}
+
+	// The initial source range of 127.0.0.0/8 should block all traffic.
+	// Poll for a minute and verify that we do not get an HTTP response.
+	gotResponse := false
+	err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
nit: We _want_ a timeout error in this case, so maybe we should replace `err` with `_`.

Reply:
Fair point.  I suppose `err` is redundant since we have `gotResponse`.